src/components/DonationForm/DonationForm.jsx

```diff
@@ -1,54 +1,207 @@
+    .min(5, 'ZIP Code must be exactly 5 digits')
+    .max(5, 'ZIP Code must be exactly 5 digits'),
   email1: yup.string().email('Invalid email').required('Email required'),
-  email2: yup.string().email('Invalid email').required('Email required'),
+  email2: yup
+    .string()
+    .required('Email required')
+    .oneOf([yup.ref('email1'), null], 'Emails must both match'),
+  Items: yup.array().of(yup.object().shape(itemFieldSchema), 'A Furniture Selection is Required'),
 });
 
 function DonationForm() {
   const {
+    handleSubmit,
     register,
     control,
-    handleSubmit,
     formState: { errors },
   } = useForm({
     resolver: yupResolver(schema),
   });
 
+  const [furnitureOptions, setFurnitureOptions] = useState([
+    // 'Select Furniture',
+    'Dressers',
+    'Clean Housewares',
+    'Antiques',
+    'Art',
+    'Clean rugs',
+    'Home Decor items',
+    'Pet care items',
+    'Patio Furniture',
+  ]);
+
+  const [DonatedFurnitureList, setDonatedFurniture] = useState([]);
+
+  const [selectedFurnitureValue, setSelectedFurnitureValue] = useState('');
+
+  // const {
+  //   fields: DonatedFurnitureList,
+  //   append: appendDonation,
+  //   remove: removeFurniture,
+  // } = useFieldArray({
+  //   control,
+  //   name: 'DonatedFurniture',
+  // });
+
   const {
-    fields: furnitureFields,
-    append: appendFurniture,
-    remove: removeFurniture,
+    fields: descriptionsIntermediateList,
+    update: updateDescription,
+    replace: replaceIntermediateDescriptions,
+    append: appendDescription,
+    remove: removeDescription,
   } = useFieldArray({
     control,
-    name: 'furnitureField',
+    name: 'ImageDetail',
   });
 
+  // const {
+  //   fields: itemsList,
+  //   append: appendItem,
+  //   remove: removeItem,
+  // } = useFieldArray({
+  //   control,
+  //   name: 'Items',
+  // });
+
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [files, setFiles] = useState([]);
+  const [filesIntermediate, setFilesIntermediate] = useState([]);
+  const [descriptions, setDescriptions] = useState([]);
+
+  const removeIntermediateFile = index => {
+    setFilesIntermediate(
+      filesIntermediate.filter(item => filesIntermediate.indexOf(item) !== index),
+    );
+  };
+
+  const removeIntermediateDescription = index => {
+    removeDescription(index);
+  };
+
+  const getDescription = index => {
+    const descObj = descriptionsIntermediateList[index];
+    if (descObj) {
+      return Object.values(descObj).slice(0, -1).join('');
+    }
+    return '';
+  };
+
+  const updateIntermediateDescription = (index, newDescription) => {
+    if (index >= descriptionsIntermediateList.length) {
+      appendDescription('');
+    }
+    updateDescription(index, newDescription);
+  };
+
+  const onSubmit = async data => {
+    // const urls = await Promise.all(files.map(async file => uploadImage(file)));
+    sendEmail(data.email1, dconfirmemailtemplate);
+  };
```

Comment:
When submitting the donation form, we should all the form details to the backend to create a new donation.